src/lib.rs

```diff
@@ -894,6 +909,33 @@ fn into_fr_plus_1<I: IntoFr>(x: I) -> Fr {
     result
 }
 
+fn derivation_index_into_fr(v: &[u8]) -> Fr {
+    // use number of rounds as a salt to avoid
+    // any child hash giving the same sequence
+    index_and_rounds_into_fr(v, 0)
+}
+
+fn index_and_rounds_into_fr(v: &[u8], rounds: u8) -> Fr {
+    let mut vr = vec![0u8; v.len() + 1];
+    vr[..v.len()].clone_from_slice(v);
+    vr[v.len()] = rounds;
+    let h = sha3_256(&vr);
```

Comment:
Depending on what the user is using as indices, this clone could get expensive, consider using the Hasher api directly, it was designed to avoid these intermediate buffers. Could give you a nice boost in mem/perf especially since this is called iteratively.
```suggestion
    let mut sha3 = Sha3::v256();
    sha3.update(v);
    sha3.update(rounds);
    let mut h = [0u8; 32];
    sha3.finalize(&mut h);
    h
```

Reply:
Nice tip. I've updated and force pushed with https://github.com/maidsafe/blsttc/pull/23/commits/88739ed69c3d7f4ee267f80484a1969d70051627